### PR TITLE
SPOCopilotPromoOptInStatus.md

### PR DESCRIPTION
**Pull request by siruvanti**
https://github.com/MicrosoftDocs/OfficeDocs-SharePoint-PowerShell/commit/d1857bdc13491227d23301172f7dd8fc5f380959

https://github.com/MicrosoftDocs/OfficeDocs-SharePoint-PowerShell/commit/8e1fb12030b8473419ebdf623b17e997f1ef81ee

### CODE REVIEW

File: sharepoint/sharepoint-ps/sharepoint-online/Get-SPOCopilotPromoOptInStatus.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+The `Get-SPOCopilotPromoOptInStatus` cmdlet retrieves and returns the opt-in Copilot promo status. The user must be a SharePoint Admin to run the cmlets.
+
+## EXAMPLES
+
+### Example 1
+
+```powershell
+Get-SPOCopilotPromoOptInStatusEnabled
+true
+```
```
Comment (samkabue) [inline sharepoint/sharepoint-ps/sharepoint-online/Get-SPOCopilotPromoOptInStatus.md:35]: remove this `true`

File: sharepoint/sharepoint-ps/sharepoint-online/Get-SPOCopilotPromoOptInStatus.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+Example 1: Success message shown to user when commandlet returns a positive flag from Tenant store
+
+### Example 2
+
+```powershell
+Get-SPOCopilotPromoOptInStatusDisabled
+False
+```
+
+Example 2: Success message shown to user when commandlet returns a negative flag from Tenant store
```
Comment (samkabue) [inline sharepoint/sharepoint-ps/sharepoint-online/Get-SPOCopilotPromoOptInStatus.md:46]: remove this example

File: sharepoint/sharepoint-ps/sharepoint-online/Get-SPOCopilotPromoOptInStatus.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+## RELATED LINKS
+
+Set-SPOCopilotPromoOptInStatus
```
Comment (samkabue) [inline sharepoint/sharepoint-ps/sharepoint-online/Get-SPOCopilotPromoOptInStatus.md:50]: if you include documentation for Set-SPOCopilotPromoOptInStatus in this PR, then this can be

```
[Set-SPOCopilotPromoOptInStatus](./Set-SPOCopilotPromoOptInStatus.md)
```

File: sharepoint/sharepoint-ps/sharepoint-online/Get-SPOCopilotPromoOptInStatus.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+## EXAMPLES
+
+### Example 1
+
+```powershell
+Get-SPOCopilotPromoOptInStatusEnabled
+true
+```
+
+Example 1: Success message shown to user when commandlet returns a positive flag from Tenant store
```
Comment (samkabue) [inline sharepoint/sharepoint-ps/sharepoint-online/Get-SPOCopilotPromoOptInStatus.md:37]: change to this:

`Example 1 returns the value of the Opt-In promo status.`

File: sharepoint/sharepoint-ps/sharepoint-online/Get-SPOCopilotPromoOptInStatus.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+schema: 
+author: swathi.iruvanti
+ms.author: siruvanti
+ms.reviewer:
+---
+# Get-SPOCopilotPromoOptInStatus
+
+## SYNOPSIS
+
+Returns the true or false Opt-In Copilot promo status
```
Comment (samkabue) [inline sharepoint/sharepoint-ps/sharepoint-online/Get-SPOCopilotPromoOptInStatus.md:17]: reword it to something like this:

```
This cmdlet returns the Opt-In Copilot promo status for the tenant.
If the promo status is enabled, the return value is `True`, otherwise the return value is `False`.
```

File: sharepoint/sharepoint-ps/sharepoint-online/Get-SPOCopilotPromoOptInStatus.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+Returns the true or false Opt-In Copilot promo status 
+
+## SYNTAX
+```powershell
+Get-SPOCopilotPromoOptInStatus
+```
+
+## DESCRIPTION
+
+The `Get-SPOCopilotPromoOptInStatus` cmdlet retrieves and returns the opt-in Copilot promo status. The user must be a SharePoint Admin to run the cmlets.
```
Comment (samkabue) [inline sharepoint/sharepoint-ps/sharepoint-online/Get-SPOCopilotPromoOptInStatus.md:26]: reword it to something like this:

```
This cmdlet returns the Opt-In Copilot promo status for the tenant.
If the promo status is enabled, the return value is `True`, otherwise the return value is `False`.
```

File: sharepoint/sharepoint-ps/sharepoint-online/Get-SPOCopilotPromoOptInStatus.md (hunk 3 above)
Comment (samkabue) [reply sharepoint/sharepoint-ps/sharepoint-online/Get-SPOCopilotPromoOptInStatus.md:50]: also include this one:

```
[Getting started with SharePoint Online Management Shell](/powershell/sharepoint/sharepoint-online/connect-sharepoint-online)
```

**Review by samkabue — CHANGES_REQUESTED** [review]
:clock1:

**Comment by siruvanti** [comment]
> @siruvanti please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement
> ### Contribution License Agreement
> This Contribution License Agreement (**“Agreement”**) is agreed to by the party signing below (**“You”**), and conveys certain license rights to Microsoft Corporation and its affiliates (“Microsoft”) for Your contributions to Microsoft open source projects. This Agreement is effective as of the latest signature date below.
> 
> 1. **Definitions**.
>    **“Code”** means the computer software code, whether in human-readable or machine-executable form,
>    that is delivered by You to Microsoft under this Agreement.
>    **“Project”** means any of the projects owned or managed by Microsoft and offered under a license
>    approved by the Open Source Initiative ([www.opensource.org](http://www.opensource.org)).
>    **“Submit”** is the act of uploading, submitting, transmitting, or distributing code or other content to any
>    Project, including but not limited to communication on electronic mailing lists, source code control
>    systems, and issue tracking systems that are managed by, or on behalf of, the Project for the purpose of
>    discussing and improving that Project, but excluding communication that is conspicuously marked or
>    otherwise designated in writing by You as “Not a Submission.”
>    **“Submission”** means the Code and any other copyrightable material Submitted by You, including any
>    associated comments and documentation.
> 2. **Your Submission**. You must agree to the terms of this Agreement before making a Submission to any
>    Project. This Agreement covers any and all Submissions that You, now or in the future (except as
>    described in Section 4 below), Submit to any Project.
> 3. **Originality of Work**. You represent that each of Your Submissions is entirely Your original work.
>    Should You wish to Submit materials that are not Your original work, You may Submit them separately
>    to the Project if You (a) retain all copyright and license information that was in the materials as You
>    received them, (b) in the description accompanying Your Submission, include the phrase “Submission
>    containing materials of a third party:” followed by the names of the third party and any licenses or other
>    restrictions of which You are aware, and (c) follow any other instructions in the Project’s written
>    guidelines concerning Submissions.
> 4. **Your Employer**. References to “employer” in this Agreement include Your employer or anyone else
>    for whom You are acting in making Your Submission, e.g. as a contractor, vendor, or agent. If Your
>    Submission is made in the course of Your work for an employer or Your employer has intellectual
>    property rights in Your Submission by contract or applicable law, You must secure permission from Your
>    employer to make the Submission before signing this Agreement. In that case, the term “You” in this
>    Agreement will refer to You and the employer collectively. If You change employers in the future and
>    desire to Submit additional Submissions for the new employer, then You agree to sign a new Agreement
>    and secure permission from the new employer before Submitting those Submissions.
> 5. **Licenses**.
> 
> * **Copyright License**. You grant Microsoft, and those who receive the Submission directly or
>   indirectly from Microsoft, a perpetual, worldwide, non-exclusive, royalty-free, irrevocable license in the
>   Submission to reproduce, prepare derivative works of, publicly display, publicly perform, and distribute
>   the Submission and such derivative works, and to sublicense any or all of the foregoing rights to third
>   parties.
> * **Patent License**. You grant Microsoft, and those who receive the Submission directly or
>   indirectly from Microsoft, a perpetual, worldwide, non-exclusive, royalty-free, irrevocable license under
>   Your patent claims that are necessarily infringed by the Submission or the combination of the
>   Submission with the Project to which it was Submitted to make, have made, use, offer to sell, sell and
>   import or otherwise dispose of the Submission alone or with the Project.
> * **Other Rights Reserved**. Each party reserves all rights not expressly granted in this Agreement.
>   No additional licenses or rights whatsoever (including, without limitation, any implied licenses) are
>   granted by implication, exhaustion, estoppel or otherwise.
> 
> 6. **Representations and Warranties**. You represent that You are legally entitled to grant the above
>    licenses. You represent that each of Your Submissions is entirely Your original work (except as You may
>    have disclosed under Section 3). You represent that You have secured permission from Your employer to
>    make the Submission in cases where Your Submission is made in the course of Your work for Your
>    employer or Your employer has intellectual property rights in Your Submission by contract or applicable
>    law. If You are signing this Agreement on behalf of Your employer, You represent and warrant that You
>    have the necessary authority to bind the listed employer to the obligations contained in this Agreement.
>    You are not expected to provide support for Your Submission, unless You choose to do so. UNLESS
>    REQUIRED BY APPLICABLE LAW OR AGREED TO IN WRITING, AND EXCEPT FOR THE WARRANTIES
>    EXPRESSLY STATED IN SECTIONS 3, 4, AND 6, THE SUBMISSION PROVIDED UNDER THIS AGREEMENT IS
>    PROVIDED WITHOUT WARRANTY OF ANY KIND, INCLUDING, BUT NOT LIMITED TO, ANY WARRANTY OF
>    NONINFRINGEMENT, MERCHANTABILITY, OR FITNESS FOR A PARTICULAR PURPOSE.
> 7. **Notice to Microsoft**. You agree to notify Microsoft in writing of any facts or circumstances of which
>    You later become aware that would make Your representations in this Agreement inaccurate in any
>    respect.
> 8. **Information about Submissions**. You agree that contributions to Projects and information about
>    contributions may be maintained indefinitely and disclosed publicly, including Your name and other
>    information that You submit with Your Submission.
> 9. **Governing Law/Jurisdiction**. This Agreement is governed by the laws of the State of Washington, and
>    the parties consent to exclusive jurisdiction and venue in the federal courts sitting in King County,
>    Washington, unless no federal subject matter jurisdiction exists, in which case the parties consent to
>    exclusive jurisdiction and venue in the Superior Court of King County, Washington. The parties waive all
>    defenses of lack of personal jurisdiction and forum non-conveniens.
> 10. **Entire Agreement/Assignment**. This Agreement is the entire agreement between the parties, and
>     supersedes any and all prior agreements, understandings or communications, written or oral, between
>     the parties relating to the subject matter hereof. This Agreement may be assigned by Microsoft.

**Comment by siruvanti** [comment]
CLA: 
@microsoft-github-policy-service agree [company="{your company}"]

File: sharepoint/sharepoint-ps/sharepoint-online/Get-SPOCopilotPromoOptInStatus.md [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+---
+- External help file: sharepointonline.xml
+- Module Name: Microsoft.Online.SharePoint.PowerShell
+- Online version: https://learn.microsoft.com/powershell/module/sharepoint-online/Get-SPOCopilotPromoOptInStatus
+- Applicable: SharePoint Online
+- Title: Get-SPOCopilotPromoOptInStatus.md
+- Schema: 
+- Author: swathi.iruvanti
+- Ms.author: siruvanti
+- Ms.reviewer:
```
Comment (samkabue) [inline sharepoint/sharepoint-ps/sharepoint-online/Get-SPOCopilotPromoOptInStatus.md:11]: fix these. see your other file

File: sharepoint/sharepoint-ps/sharepoint-online/Get-SPOCopilotPromoOptInStatus.md [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+Get-SPOCopilotPromoOptInStatusEnabled
+```
+
+Example 1: returns the value of the Opt-In promo status.
+
+## RELATED LINKS
+
+- [Set-SPOCopilotPromoOptInStatus](./Set-SPOCopilotPromoOptInStatus.md)
+
+
```
Comment (samkabue) [inline sharepoint/sharepoint-ps/sharepoint-online/Get-SPOCopilotPromoOptInStatus.md:44]: remove blank lines

File: sharepoint/sharepoint-ps/sharepoint-online/Get-SPOCopilotPromoOptInStatus.md [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+## EXAMPLES
+
+### Example 1
+
+```powershell
+Get-SPOCopilotPromoOptInStatusEnabled
+```
+
+Example 1: returns the value of the Opt-In promo status.
```
Comment (samkabue) [inline sharepoint/sharepoint-ps/sharepoint-online/Get-SPOCopilotPromoOptInStatus.md:38]: >: [](http://example.com/codeflow?start=9&length=1)

remove the ":"

"Example 1 returns ..."

File: sharepoint/sharepoint-ps/sharepoint-online/Get-SPOCopilotPromoOptInStatus.md [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+Returns the Opt-In Copilot promo status for the tenant.
+
+## SYNTAX
+```powershell
+Get-SPOCopilotPromoOptInStatus
+```
+
+## DESCRIPTION
+
+The `Get-SPOCopilotPromoOptInStatus` This cmdlet returns the Opt-In Copilot promo status for the tenant.
```
Comment (samkabue) [inline sharepoint/sharepoint-ps/sharepoint-online/Get-SPOCopilotPromoOptInStatus.md:26]: >The `Get-SPOCopilotPromoOptInStatus`  [](http://example.com/codeflow?start=0&length=37)

remove this sentence

File: sharepoint/sharepoint-ps/sharepoint-online/Set-SPOCopilotPromoOptInStatus.md [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+schema: 
+author: swathi.iruvanti
+ms.author: siruvanti
+ms.reviewer:
+---
+# Set-SPOCopilotPromoOptInStatus
+
+## SYNOPSIS
+
+Retrieves the IsCopilotPromoStatusEnabled ( "true/false" ) state of the Copilot promo opt-in bit and stores the information.
```
Comment (samkabue) [inline sharepoint/sharepoint-ps/sharepoint-online/Set-SPOCopilotPromoOptInStatus.md:16]: this cmdlet does "set" -- it doesn't "retrieve" -- so change this to something like

```
Sets the Opt-In Copilot promo status for the tenant.
```

File: sharepoint/sharepoint-ps/sharepoint-online/Set-SPOCopilotPromoOptInStatus.md [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+```
+
+### Parameters:
+
+The following details are returned:
+
+- Copilot promo status set successfully, returns nothing if successful. 
+
+- Copilot promo status set failed, returns an error. 
+
```
Comment (samkabue) [inline sharepoint/sharepoint-ps/sharepoint-online/Set-SPOCopilotPromoOptInStatus.md:30]: this is wrong. these are not parameters.

File: sharepoint/sharepoint-ps/sharepoint-online/Set-SPOCopilotPromoOptInStatus.md [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+## DESCRIPTION
+
+The `Set-SPOCopilotPromoOptInStatus` cmdlet stores the opt-in state. The user must be a SharePoint Admin to run the cmlets.
```
Comment (samkabue) [inline sharepoint/sharepoint-ps/sharepoint-online/Set-SPOCopilotPromoOptInStatus.md:33]: move it up to after SYNTAX

File: sharepoint/sharepoint-ps/sharepoint-online/Set-SPOCopilotPromoOptInStatus.md [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+---
+# Set-SPOCopilotPromoOptInStatus
+
+## SYNOPSIS
+
+Retrieves the IsCopilotPromoStatusEnabled ( "true/false" ) state of the Copilot promo opt-in bit and stores the information.
+
+## SYNTAX
+```powershell
+Set-SPOCopilotPromoOptInStatus
```
Comment (samkabue) [inline sharepoint/sharepoint-ps/sharepoint-online/Set-SPOCopilotPromoOptInStatus.md:20]: this is missing the param.

File: sharepoint/sharepoint-ps/sharepoint-online/Set-SPOCopilotPromoOptInStatus.md (hunk 13 above)
Comment (samkabue) [inline sharepoint/sharepoint-ps/sharepoint-online/Set-SPOCopilotPromoOptInStatus.md:33]: reword to something like:

```
This cmdlet sets the Opt-In Copilot promo status for the tenant to `True` or `False`.
```

File: sharepoint/sharepoint-ps/sharepoint-online/Set-SPOCopilotPromoOptInStatus.md [hunk 15]
```diff
@@ -0,0 +1,59 @@
...
+
+The `Set-SPOCopilotPromoOptInStatus` cmdlet stores the opt-in state. The user must be a SharePoint Admin to run the cmlets.
+
+
+## EXAMPLES
+
+### Example 1
+
+```powershell
+Set-SPOCopilotPromoOptInStatusSetSuccessfully -SPOCopilotPromoOptInStatusEnabled true
```
Comment (samkabue) [inline sharepoint/sharepoint-ps/sharepoint-online/Set-SPOCopilotPromoOptInStatus.md:41]: >true [](http://example.com/codeflow?start=81&length=4)

```
$true
```

File: sharepoint/sharepoint-ps/sharepoint-online/Set-SPOCopilotPromoOptInStatus.md [hunk 16]
```diff
@@ -0,0 +1,59 @@
...
+If commandlet fails to execute for Set-CopilotPromoOptInStatus, an error message is shown to user.
+
+### Example 2
+
+```powershell
+Set-SPOCopilotPromoOptInStatusSetSuccessfully -SPOCopilotPromoOptInStatusDisabled false
+```
+
+Example 2: Success message shown to user when commandlet executes successfully for Set-CopilotPromoOptInStatus
+If commandlet fails to execute for Set-CopilotPromoOptInStatus, an error message is shown to user.
```
Comment (samkabue) [inline sharepoint/sharepoint-ps/sharepoint-online/Set-SPOCopilotPromoOptInStatus.md:54]: remove example 2 -- superfluous and not needed

File: sharepoint/sharepoint-ps/sharepoint-online/Set-SPOCopilotPromoOptInStatus.md [hunk 17]
```diff
@@ -0,0 +1,59 @@
...
+## EXAMPLES
+
+### Example 1
+
+```powershell
+Set-SPOCopilotPromoOptInStatusSetSuccessfully -SPOCopilotPromoOptInStatusEnabled true
+```
+
+Example 1: No success message returned, can validate by executing Get-SPOCopilotPromoOptInStatus.
+If commandlet fails to execute for Set-CopilotPromoOptInStatus, an error message is shown to user.
```
Comment (samkabue) [inline sharepoint/sharepoint-ps/sharepoint-online/Set-SPOCopilotPromoOptInStatus.md:45]: unnecessary details. reword to something like

```
Example 1 sets the Opt-In Copilot promo status for the tenant to `True`.
```

File: sharepoint/sharepoint-ps/sharepoint-online/Set-SPOCopilotPromoOptInStatus.md [hunk 18]
```diff
@@ -0,0 +1,59 @@
...
+Set-SPOCopilotPromoOptInStatusSetSuccessfully -SPOCopilotPromoOptInStatusDisabled false
+```
+
+Example 2: Success message shown to user when commandlet executes successfully for Set-CopilotPromoOptInStatus
+If commandlet fails to execute for Set-CopilotPromoOptInStatus, an error message is shown to user.
+
+
+## RELATED LINKS
+
+Get-SPOCopilotPromoOptInStatus
```
Comment (samkabue) [inline sharepoint/sharepoint-ps/sharepoint-online/Set-SPOCopilotPromoOptInStatus.md:59]: this is not correct way to link -- see other file for my comments

File: sharepoint/sharepoint-ps/sharepoint-online/Set-SPOCopilotPromoOptInStatus.md [hunk 19]
```diff
@@ -0,0 +1,53 @@
...
+---
+# Set-SPOCopilotPromoOptInStatus
+
+## SYNOPSIS
+
+Sets the Opt-In Copilot promo status for the tenant.
+
+## SYNTAX
+```powershell
+Set-SPOCopilotPromoOptInStatus -IsCopilotPromoStatusEnabled $true
```
Comment (samkabue) [inline sharepoint/sharepoint-ps/sharepoint-online/Set-SPOCopilotPromoOptInStatus.md:20]: >$true [](http://example.com/codeflow?start=60&length=5)

should be

```
<Boolean>
```

File: sharepoint/sharepoint-ps/sharepoint-online/Set-SPOCopilotPromoOptInStatus.md [hunk 20]
```diff
@@ -0,0 +1,53 @@
...
+```
+## DESCRIPTION
+
+This cmdlet sets the Opt-In Copilot promo status for the tenant to `True` or `False`. The user must be a SharePoint Admin to run the cmlets.
```
Comment (samkabue) [inline sharepoint/sharepoint-ps/sharepoint-online/Set-SPOCopilotPromoOptInStatus.md:24]: > The user must be a SharePoint Admin to run the cmlets. [](http://example.com/codeflow?start=85&length=55)

remove this sentence as it doesn't add any value.

SPO PowerShell cmdlets can only be called by SPO admins

File: sharepoint/sharepoint-ps/sharepoint-online/Set-SPOCopilotPromoOptInStatus.md [hunk 21]
```diff
@@ -0,0 +1,53 @@
...
+Type: boolean
+Parameter Sets: 
+Aliases:
+Applicable: SharePoint Online
+
+Required: True
+Default value: None
+```
+## RELATED LINKS
+	- [Get-SPOCopilotPromoOptInStatus](./Get-SPOCopilotPromoOptInStatus.md)
```
Comment (samkabue) [inline sharepoint/sharepoint-ps/sharepoint-online/Set-SPOCopilotPromoOptInStatus.md:52]: >	 [](http://example.com/codeflow?start=0&length=1)

remove this tab indent

same to the next line

File: sharepoint/sharepoint-ps/sharepoint-online/Set-SPOCopilotPromoOptInStatus.md [hunk 22]
```diff
@@ -0,0 +1,54 @@
+---
+external help file: sharepointonline.xml
+Module Name: Microsoft.Online.SharePoint.PowerShell
+online version: https://learn.microsoft.com/powershell/module/sharepoint-online/Set-SPOCopilotPromoOptInStatus
+applicable: SharePoint Online
+title: Set-SPOCopilotPromoOptInStatus.md
```
Comment (samkabue) [inline sharepoint/sharepoint-ps/sharepoint-online/Set-SPOCopilotPromoOptInStatus.md:6]: >.md [](http://example.com/codeflow?start=37&length=3)

remove ".md" extension

File: sharepoint/sharepoint-ps/sharepoint-online/Get-SPOCopilotPromoOptInStatus.md [hunk 23]
```diff
@@ -0,0 +1,41 @@
+---
+external help file: sharepointonline.xml
+Module Name: Microsoft.Online.SharePoint.PowerShell
+online version: https://learn.microsoft.com/powershell/module/sharepoint-online/Get-SPOCopilotPromoOptInStatus
+applicable: SharePoint Online
+title: Get-SPOCopilotPromoOptInStatus.md
```
Comment (samkabue) [inline sharepoint/sharepoint-ps/sharepoint-online/Get-SPOCopilotPromoOptInStatus.md:6]: >.md [](http://example.com/codeflow?start=37&length=3)

remove ".md" extension

**Review by samkabue — APPROVED** [review]
:shipit:
